test/integration/client/client_test.go

```diff
@@ -882,3 +884,122 @@ func TestApplyWithApplyConfiguration(t *testing.T) {
+	if err != nil {
+		t.Fatalf("Error creating createApply: %v", err)
+	}
+
+	// apply some non-overlapping fields with another field manager
+	sidecarApply := appsv1ac.Deployment("nginx-apply", "default").
+		WithSpec(appsv1ac.DeploymentSpec().
+			WithTemplate(corev1ac.PodTemplateSpec().
+				WithSpec(corev1ac.PodSpec().
+					WithContainers(
+						corev1ac.Container().
+							WithName("sidecar").
+							WithImage("nginx:1.14.2"),
+					),
+				),
+			),
+		)
+
+	applied, err := deploymentClient.Apply(context.TODO(), sidecarApply, metav1.ApplyOptions{FieldManager: "sidecar-mgr", Force: true})
+	if err != nil {
+		t.Fatalf("Error applying createApply: %v", err)
+	}
+	sidecarExtracted, err := appsv1ac.ExtractDeployment(applied, "sidecar-mgr")
+	if err != nil {
+		t.Fatalf("Error extracting createApply apply configuration: %v", err)
+	}
+	if !equality.Semantic.DeepEqual(sidecarApply, sidecarExtracted) {
+		t.Errorf("Expected sidecarExtracted apply configuration to match original, but got:\n%s\n", cmp.Diff(sidecarApply, sidecarExtracted))
+	}
+
+	// modify the extracted apply configuration that was just applied and add some fields that overlap
+	// with the fields owned by the other field manager to force ownership of them
+	sidecarExtracted.Spec.Template.Spec.Containers[0].WithImage("nginx:1.14.3")
+	sidecarExtracted.Spec.Template.Spec.WithContainers(corev1ac.Container().
+		WithName("nginx").
+		WithWorkingDir("/tmp/v2"),
+	)
+	reapplied, err := deploymentClient.Apply(context.TODO(), sidecarExtracted, metav1.ApplyOptions{FieldManager: "sidecar-mgr", Force: true})
+	if err != nil {
+		t.Fatalf("Unexpected error when applying manifest for Deployment: %v", err)
+	}
+
+	// extract apply configurations for both field managers and check that they are what we expect
+	reappliedExtracted, err := appsv1ac.ExtractDeployment(reapplied, "sidecar-mgr")
+	if err != nil {
+		t.Fatalf("Error extracting sidecarExtracted apply configuration: %v", err)
+	}
+
+	expectedReappliedExtracted := appsv1ac.Deployment("nginx-apply", "default").
+		WithSpec(appsv1ac.DeploymentSpec().
+			WithTemplate(corev1ac.PodTemplateSpec().
+				WithSpec(corev1ac.PodSpec().
+					WithContainers(
+						corev1ac.Container().
+							WithName("nginx").
+							WithWorkingDir("/tmp/v2"),
+						corev1ac.Container().
+							WithName("sidecar").
+							WithImage("nginx:1.14.3"),
+					),
+				),
+			),
+		)
+	if !equality.Semantic.DeepEqual(expectedReappliedExtracted, reappliedExtracted) {
+		t.Errorf("Reapplied apply configuration did not match expected, got:\n%s\n", cmp.Diff(expectedReappliedExtracted, reappliedExtracted))
+	}
+
+	createMgrExtracted, err := appsv1ac.ExtractDeployment(reapplied, "create-mgr")
+	if err != nil {
+		t.Fatalf("Error extracting createApply apply configuration: %v", err)
+	}
+
+	expectedCreateExtracted := appsv1ac.Deployment("nginx-apply", "default").
```

Comment:
As mentioned in the slack, these tests should cover at least:
1. changing single element, replicas, for example, my example from https://gist.github.com/soltysh/7f5e924e6dee5e69a1ab00c6a72012b1#file-test3-go-L82 fails with validation error b/c labels are removed.
2. updating status (this is what majority of controllers does).

Reply:
Thank

> > The alternatives were: (1) embed it as a string, (2) download/cache it from the api server or (3) move the extract logic server side. #1 was the only option that seemed tractable in the short term.
> 
> This seems reasonable for generated clients. I'd like to see a way to use this with a dynamic client in the next release. I suspect that will mean finding a way to get the required schema from the openapi doc.

Yeah, @lavalamp had mentioned needing to support the dynamic client as well. We'll follow up.

Reply:
I've added a bunch more test cases to both cover these cases and to improve how well the tests document the functionality.